src/app/api/v1/endpoints/create_pngs.py

```diff
@@ -0,0 +1,74 @@
+"""This module defines the FastAPI endpoint for converting a PDF document to PNG images."""
+
+import os
+
+import fitz
+from app.common.aws import load_pdf_from_aws, s3_client
+from app.common.config import config
+from app.common.schemas import PNGResponse
+from fastapi import Form, HTTPException
+
+
+def create_pngs(aws_filename: str = Form(...)):
+    """Convert a PDF document to PNG images. Please note that this function will overwrite any existing PNG files.
+
+    Args:
+        aws_filename (str): The name of the PDF document in the S3 bucket. For example, "pdfs/geoquat/train/10012.pdf".
+
+    Returns:
+        PNGResponse: The URLs of the PNG images in the S3 bucket.
+    """
+    # Validate the filename parameter
+    if not aws_filename or not isinstance(aws_filename, str):
+        raise HTTPException(
+            status_code=400, detail="Invalid request. 'filename' parameter is required and must be a string."
+        )
+
+    # Check if the PDF name is valid
+    if not aws_filename.endswith(".pdf"):
+        raise HTTPException(status_code=400, detail="Invalid request. The filename must end with '.pdf'.")
+    if len(aws_filename.split("/")) < 4:
+        error_details = "The filename must be in the format 'pdfs/{dataset_name}/{dataset_type}/{filename}.pdf'."
+        raise HTTPException(
+            status_code=400,
+            detail=error_details,
+        )
```

Comment:
We do not have to require this. I do believe that there is a misunderstanding on my side since I tried to replicate the structure we currently have for the data. So, just to be sure, the S3 bucket should have the `/pdf` and `/png' folders, but we do not split the data into train, validation, and test? This is indeed simpler. 

Regarding the prefixes, we can indeed automatically add them.